lib/ansible/modules/windows/win_domain_computer.ps1

```diff
@@ -0,0 +1,183 @@
+  # Test computer exists
+  $computer = Try { Get-ADComputer `
+      -Identity $desired_state.name `
+      -Properties DistinguishedName,DNSHostName,Enabled,Name,SamAccountName,Description,ObjectClass
+    } Catch { $null }
+  If ($computer) {
+      $initial_state = @{
+      name = $computer.Name
+      sam_account_name = $computer.SamAccountName
+      dns_hostname = $computer.DNSHostName
+      # Get OU from regexp that removes all characters to the first ","
+      ou = $computer.DistinguishedName -creplace "^[^,]*,",""
+      distinguished_name = $computer.DistinguishedName
+      description = $computer.Description
+      enabled = $computer.Enabled
+      state = "present"
+    }
+  } Else {
+    $initial_state = @{
+      name = $desired_state.name
+      state = "absent"
+    }
+  }
+
+  return $initial_state
+}
+
+# ------------------------------------------------------------------------------
+Function Set-ConstructedState($initial_state, $desired_state) {
+  Try {
+    Set-ADComputer `
+      -Identity $desired_state.name `
+      -SamAccountName $desired_state.name `
+      -DNSHostName $desired_state.dns_hostname `
+      -Enabled $desired_state.enabled `
+      -Description $desired_state.description `
+      -WhatIf:$check_mode
+  } Catch { Fail-Json $result "Error in function Set-ConstructedState" }
+
+  If ($initial_state.distinguished_name -ne $desired_state.distinguished_name)
+  {
+    # Move computer to OU
+    Try {
+      Get-ADComputer $desired_state.name |
+          Move-ADObject `
+            -TargetPath $desired_state.ou `
+            -Confirm:$False `
+            -WhatIf:$check_mode
+    } Catch { Fail-Json $result "Error in function Move-ADObject" }
+  }
+  Set-Attr $result "changed" $true
+}
+
+# ------------------------------------------------------------------------------
+Function Add-ConstructedState($desired_state) {
+  Try {
+    New-ADComputer `
+      -Name $desired_state.name `
+      -SamAccountName $desired_state.sam_account_name `
+      -DNSHostName $desired_state.dns_hostname `
+      -Path $desired_state.ou `
+      -Enabled $desired_state.enabled `
+      -Description $desired_state.description `
+      -WhatIf:$check_mode
+  } Catch { Fail-Json $result "Error in function Add-ConstructedState" }
+
+  Set-Attr $result "changed" $true
+}
+
+# ------------------------------------------------------------------------------
+Function Remove-ConstructedState($initial_state) {
+  Try {
+    Remove-ADComputer `
+      -Identity $initial_state.name `
+      -Confirm:$False `
+      -WhatIf:$check_mode
+  } Catch { Fail-Json $result "Error in function Remove-ADComputer" }
+
+  Set-Attr $result "changed" $true
+}
+
+# ------------------------------------------------------------------------------
+Function are_hashtables_equal($x, $y) {
+  # Compare not nested HashTables
+  Foreach ($key in $x.Keys) {
+      If (($y.Keys -notcontains $key) -or ($x[$key] -ne $y[$key])) {
```

Comment:
You need to be carefule with this, by default the comparisons are not case sensitive. If we want to compare equality based on case you need to use `-cne`